corral/bundle/bundle.pony

```diff
@@ -9,30 +10,30 @@ class Bundle
   those files.
   """
   let dir: FilePath
-  let log: Log
+  let log: Logger[String]
   let info: InfoData
   let deps: Map[String, Dep ref] = deps.create()
   var modified: Bool = false
 
-  new iso create(dir': FilePath, log': Log) =>
+  new iso create(dir': FilePath, log': Logger[String]) =>
     dir = dir'
     log = log'
     info = InfoData(JsonObject)
-    log.info("Created bundle in " + dir.path)
+    log(Info) and log.log("Created bundle in " + dir.path)
```

Comment:
here's a question, should there be a "Log" class for corral that keeps the previous form of different methods and it contains the Logger actor and the check to see whether to send a message? It's certainly cleaner and hides to `log(Info)`.

By hiding that, we could have single comments per method on the Logger class that explains why that is done to short circuit sending a message to the actor.

It might make the code base more approachable for folks who don't really know Pony.

Thoughts?

I'm not sure how that would come together with the other changes. Perhaps it might be problematic. I'm not sure.

Reply:
I looked into that approach, but when I read the docs for the logger package and noticed the use of boolean short-circuiting to avoid the work of building strings, I didn't see a clear way to preserve the encapsulation provided by Log and to get the near zero-cost logging offered by the logger API. That's why I chose to replace all the call sites with the idiomatic use of logger.

Reply:
Makes sense.

Reply:
It's amusing that you are reminding me of this given that I wrote those docs for the logger package after writing the logger package.

😆